### PR TITLE
Added new_with_page_meta to PageReader

### LINKED ISSUE #135: Pass only necessary data when create PageReader
The new method of `PageReader` needs `&ColumnChunkMetaData` while it only use a small part of input.
 
In most scenarios, it is not the problem.
https://github.com/jorgecarleitao/parquet2/blob/fa6fa3ca3848c29d8efa80fbf42ee6a5a58cb077/src/read/page/reader.rs#L53-L58

However, user may not invoke `read_metadata()`before read page data, such as using the cached data, which is a necessary part of `ColumnChunkMetaData`. Then it is not easy to create a `PageReader`, because `ColumnChunkMetaData` is huge, and `column_chunk` is even in another crate.

How about adding a new constructor for `PageReader`, like: 
```rust
pub struct PageMetaData {
    pub num_values: i64,
    pub compression: Compression,
    pub descriptor: ColumnDescriptor,
}
pub fn new_with_page_meta(
        reader: R,
        column: PageMetaData,
        pages_filter: PageFilter,
        buffer: Vec<u8>,
    ) -> Self {
        Self {
            reader,
            total_num_values: column.num_values,
            compression: column.compression,
            seen_num_values: 0,
            current_dictionary: None,
            descriptor: column.descriptor.descriptor,
            pages_filter,
            buffer,
        }
    }
```

### PR DESCRIPTION
**Pull request by ygf11**
closes #135

### CODE REVIEW

File: src/read/page/reader.rs [hunk 1]
```diff
@@ -15,6 +15,28 @@ use crate::page::{
 
 use super::PageIterator;
 
+/// This meta is a small part of [`ColumnChunkMetaData`].
+#[derive(Debug, Clone)]
+pub struct PageMetaData {
+    // The number of values in this column chunk.
```
Comment (jorgecarleitao) [inline src/read/page/reader.rs:21]:
```suggestion
    /// The number of values in this column chunk.
```

File: src/read/page/reader.rs [hunk 2]
```diff
@@ -15,6 +15,28 @@ use crate::page::{
...
+    pub num_values: i64,
+    // Compression type
```
Comment (jorgecarleitao) [inline src/read/page/reader.rs:23]:
```suggestion
    /// Compression type
```

File: src/read/page/reader.rs [hunk 3]
```diff
@@ -15,6 +15,28 @@ use crate::page::{
...
+    pub compression: Compression,
+    // The descriptor of this parquet column
```
Comment (jorgecarleitao) [inline src/read/page/reader.rs:25]:
```suggestion
    /// The descriptor of this parquet column
```

File: src/read/page/reader.rs [hunk 4]
```diff
@@ -15,6 +15,28 @@ use crate::page::{
 
 use super::PageIterator;
 
+/// This meta is a small part of [`ColumnChunkMetaData`].
+#[derive(Debug, Clone)]
```
Comment (jorgecarleitao) [inline src/read/page/reader.rs:19]:
```suggestion
#[derive(Debug, Clone, PartialEq, Eq)]
```

File: src/read/page/reader.rs [hunk 5]
```diff
@@ -15,6 +15,28 @@ use crate::page::{
...
+impl PageMetaData {
+    /// Returns a new [`PageMetaData`].
+    pub fn new(num_values: i64, compression: Compression, descriptor: Descriptor) -> Self {
+        Self {
+            num_values,
+            compression,
+            descriptor,
+        }
+    }
+}
```
Comment (jorgecarleitao) [inline src/read/page/reader.rs:38]: I think that we should implement `From<&ColumnChunkMetadata> for PageMetaData` and reimplement `PageReader::new` as `PageReader::new_with_page_meta(reader, column.into(), pages_filter, buffer)` - this is DRY and ensures that this new code path is tested.

What do you think?

**Review by jorgecarleitao — COMMENTED** [review]
Thanks for the PR! Looks great. I left some comments on it. 

I think we should introduce this for the other page readers (`async` and filtered), so that we have a consistent API across them.

File: src/read/page/reader.rs (hunk 5 above)
Comment (ygf11) [reply src/read/page/reader.rs:38]: Agree:)

**Comment by ygf11** [comment]
> I think we should introduce this for the other page readers (async and filtered), so that we have a consistent API across them.

Yes, they should be consistent. 

Is there anything else besides `get_page_stream` and `IndexedPageReader`?

**Comment by jorgecarleitao** [comment]
> Is there anything else besides `get_page_stream` and `IndexedPageReader`?

Note, that is it 👍

File: src/read/page/reader.rs (hunk 5 above)
Comment (ygf11) [reply src/read/page/reader.rs:46]: Addressed.

Comment (ygf11) [reply src/read/page/reader.rs:46]: Addressed.

**Comment by jorgecarleitao** [comment]
thanks a lot again!

**Pull request merged**